### ISSUE TITLE
make unexpected beforeSend behaviour clear

### ISSUE THREAD

**Issue by LarsWiegers** [getsentry/sentry-docs#5988]
### Core or SDK?

Platform/SDK

### Which part? Which one?

Javascript sdk

### Description

I expected sentry to still send to error to its server if beforeSent does not return anything. 

I would expect the below to still send the error
```
BeforeSend(event, hint) {
}
```

I would expect the below to still send the error
```
BeforeSend(event, hint) {
  return event;
}
```
I would **not** expect the below to send the error
```
BeforeSend(event, hint) {
  return null;
}
```

Love Sentry btw! this was just confusing

### Suggested Solution

Either we should make this more clear in the documentation or maybe change the sdk to better make sense :)

**Comment by smeubank** [comment]
could you check this issue and see if you might have a similar issue? 

> Do you by any chance happen to have anything regarding integrations in your Sentry.init() that might be causing InboundFilters not to be enabled?

**Comment by LarsWiegers** [comment]
This is our current config (with dsn removed for security): 
init({
    app,
    logErrors: true,
    dsn: '',
    integrations: [
      new BrowserTracing({
        beforeNavigate: (context) => {
          const route = router.resolve(window.location.pathname);
          return {
            ...context,
            // You could use your UI's routing library to find the matching
            // route template here. We don't have one right now, so do some basic
            // parameter replacements.
            name: route?.resolved?.name || window.location.pathname,
          };
        },
      }),
    ],
    environment: STAGE_ENV,
    release: SENTRY_RELEASE,
    sampleRate: 0.1,
    tracesSampleRate: 0.1,
    maxBreadcrumbs: 50,
    beforeSend(event, hint?) {
      const exception = hint?.originalException;
      const message = (typeof exception === 'string' ? exception : exception?.message) ?? '';
      if (message.includes('Failed to fetch dynamically imported module')) {
        // eslint-disable-next-line no-alert
        alert(
          'It looks like you have an outdated version of the application.\n' +
            'The page must be reloaded.\n' +
            "If that doesn't help, try clearing the browser cache.",
        );
        window.location.reload();

        return null;
      }

      return event;
    },
  });

**Comment by mydea** [comment]
You are right that not returning anything from `beforeSend` will mean you don't send any event. I guess the docs can be clearer on that point, I'll write a clarification! Basically, you need to return either `null` (=nothing is sent) _or_ the actual event payload, no other response is actually valid. If you enable debug mode `debug: true` in your `init()` config, you will actually see warnings for this.

**Comment by mydea** [comment]
I've extended the docs to be more explicit and make it (hopefully) clear that `void` is not a valid return value! Thanks for reporting this.

**Comment by LarsWiegers** [comment]
Awesome @mydea thanks!
